projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import java.security.InvalidParameterException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Handles operations related to supported python versions for data job deployments. The class
+ * contains utility methods which provide functionality to read the configuration related to the
+ * python versions supported by the Control Service. These utility methods are meant to be used in
+ * other components, as needed.
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SupportedPythonVersions {
+
+  @Value("#{${datajobs.deployment.supportedPythonVersions}}")
+  private Map<String, Map<String, String>> supportedPythonVersions;
+
+  @Value("${datajobs.deployment.defaultPythonVersion}")
+  private String defaultPythonVersion;
+
+  private static final String VDK_IMAGE = "vdkImage";
+
+  private static final String BASE_IMAGE = "baseImage";
+
+  /**
+   * Check if the pythonVersion passed by the user is supported by the Control Service.
+   *
+   * @param pythonVersion python version passed by the user.
+   * @return true if the version is supported, and false otherwise.
+   */
+  public boolean isPythonVersionSupported(String pythonVersion) {
+    if (supportedPythonVersions != null) {
+      return supportedPythonVersions.containsKey(pythonVersion);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Returns a list of the python versions supported by the Control Service, in the format: [3.7,
+   * 3.8, ...]. If the supportedPythonVersions configuration is not set, the method returns the
+   * default python version set in the defaultPythonVersion configuration property.
+   *
+   * @return A list of all python versions supported by the Control Service.
+   */
+  public List<String> getSupportedPythonVersions() {
+    if (supportedPythonVersions != null && !supportedPythonVersions.isEmpty()) {
```

Comment:
again here I think that the map should only be able to be empty or non empty. never both.

Reply:
Refactored, thanks